### PR TITLE
[Roberta] fix hard wired pad token id

### LINKED ISSUE #3937: What should I do if I want to change the padding idx in pytorch bert(Huggingface)?
I find ,in RobertaModel class, the padding idx is limited to 1(is it true?),which is different from my tokenizer and data.So what should I do to change or just apply the model structure...

###
class RobertaEmbeddings(BertEmbeddings):
    """
    Same as BertEmbeddings with a tiny tweak for positional embeddings indexing.
    """

    def __init__(self, config):
        super().__init__(config)
        self.padding_idx = 1
        self.word_embeddings = nn.Embedding(config.vocab_size, config.hidden_size, padding_idx=self.padding_idx)
        self.position_embeddings = nn.Embedding(
            config.max_position_embeddings, config.hidden_size, padding_idx=self.padding_idx
        )

    def forward(self, input_ids=None, token_type_ids=None, position_ids=None, inputs_embeds=None):
        if position_ids is None:
            if input_ids is not None:
                # Create the position ids from the input token ids. Any padded tokens remain padded.
                position_ids = create_position_ids_from_input_ids(input_ids, self.padding_idx).to(input_ids.device)
            else:
                position_ids = self.create_position_ids_from_inputs_embeds(inputs_embeds)

        return super().forward(
            input_ids, token_type_ids=token_type_ids, position_ids=position_ids, inputs_embeds=inputs_embeds
        )

### PR DESCRIPTION
**Pull request by patrickvonplaten**
This might break backward compatibility for users that have uploaded a roberta model and use a different pad token than 1 in the configs.

### CODE REVIEW

**Review by julien-c — APPROVED** [review]
I think that's the right thing to do

**Comment by patrickvonplaten** [comment]
I checked all models on `https://huggingface.co/models?search=roberta` and all of them have the pad_token_id set to `1` which was used as a default before. So this is good to merge for me.
